docker_cli/image.py

```diff
@@ -0,0 +1,430 @@
+            a file. Defaults to None.
+        network : str
+            The name of the network associated with the docker image to be
+            built. Defaults to "host"
+        no-cache: bool
+            A boolean designating whether or not the docker build should use
+            the cache
+
+        Returns
+        -------
+        Image
+            An Image class instance that references a Docker image built from
+            the Dockerfile at context/dockerfile
+
+        Raises
+        -------
+        CalledProcessError
+            Raised if the docker build command fails. Holds the value returned
+            by the docker build command in its returncode attribute, if not 0.
+        """
+        ...
+
+    @overload
+    @classmethod
+    def build(
+        cls: Type[Self],
+        context: os.PathLike[str],
+        tag: str,
+        *,
+        dockerfile_string: str,
+        output_file: Optional[io.TextIOBase],
+        network: str,
+        no_cache: bool
+    ):
+        """Build a Dockerfile at the given path with the given name, then
+        return the associated Image instance.
+
+        Parameters
+        ----------
+        context : str
+            The path to the file containing the Dockerfile
+        tag : str
+            A name for the image
+        dockerfile_string : str
+            A Dockerfile-formatted string containing the information necessary
+            to build the desired image.
+        output_file : io.TextIOBase, optional
+            A file-like object that the stderr output of the docker build
+            program will be written to. If None, the method will not output to
+            a file. Defaults to None.
+        network : str
+            The name of the network associated with the docker image to be
+            built. Defaults to "host"
+        no-cache: bool
+            A boolean designating whether or not the docker build should use
+            the cache
+
+        Returns
+        -------
+        Image
+            An Image class instance that references a Docker image built from
+            the Dockerfile-formatted string at context/dockerfile_loc
+
+        Raises
+        -------
+        CalledProcessError
+            Raised if the docker build command fails. Holds the value returned
+            by the docker build command in its returncode attribute, if not 0.
+        """
+        ...
+
+    @classmethod
+    def build(
+        cls,
+        context,
+        tag,
+        *,
+        dockerfile=None,
+        dockerfile_string=None,
+        output_file=None,
+        network="host",
+        no_cache=True
+    ):
+        if dockerfile is not None:
+            dockerfile_build = True
+        elif dockerfile_string is not None:
+            dockerfile_build = False
+        else:
+            return ValueError("Image.build: either dockerfile or " +
+                              "dockerfile_string must be defined.")
```

Comment:
```suggestion
            raise ValueError("Image.build: either dockerfile or " +
                              "dockerfile_string must be defined.")
```

Reply:
No need for the `+` operator here -- consecutive string literals will concatenate in Python

https://stackoverflow.com/a/34174612

```suggestion
            raise ValueError("Image.build: either dockerfile or "
                              "dockerfile_string must be defined.")
```

Reply:
Let's remove the `"Image.build: "` part of the exception message. It's not necessary to include that info -- if an exception is raised here, users will get a traceback that includes the name of the function where the exception was raised.

```suggestion
            raise ValueError("either dockerfile or dockerfile_string must be defined.")
```